File: scalr_api_v2.py
import base64
import datetime
import hashlib
import hmac
import json
import os
import pytz
import requests
import sys

from pprint import pprint

def scalr_request_authentication (key_id, key_secret, method, path, params, body,
        date=datetime.datetime.now(tz=pytz.timezone(os.environ.get("TZ", "UTC"))).isoformat()):
    return {
        'X-Scalr-Key-Id': key_id,
        'X-Scalr-Signature': scalr_signature_algorithm(key_secret, scalr_canonical_request(method, date, path, params, body)),
        'X-Scalr-Date': date,
        'X-Scalr-Debug': 1
    }

def scalr_signature_algorithm (key_secret, canon_req):
    return 'V1-HMAC-SHA256 %s' % base64.b64encode(hmac.new(str(key_secret), canon_req, hashlib.sha256).digest())

def scalr_canonical_request (method, date, path, params, body):
    return '\n'.join([method, date, path, params, body])

def scalr_api_call (creds, path, method='GET', params='', body=''):
    return requests.get(creds.get('api_url') + path,
                            headers = scalr_request_authentication(
                                creds.get('api_key_id'), creds.get('api_key_secret'), method, path, params, body
                            )
                        )

if __name__ == "__main__":
    config = sys.argv[0].replace('.py', '.json')
    creds = json.load(open(config))
    response = scalr_api_call(creds, '/api/v1beta0/account/environments')
    pprint(json.loads(response.content))

File: scalr_api_v2.js

var crypto = require('crypto');
var format = require('string-format');
var fs = require('fs');
var request = require('request');

function scalrRequestAuthentication (keyId, keySecret, method, path, params, body, date) {
    return {
        'X-Scalr-Key-Id': keyId,
        'X-Scalr-Signature': scalrSignatureAlgorithm(keySecret, scalrCanonicalRequest(method, date, path, params, body)),
        'X-Scalr-Date': date,
        'X-Scalr-Debug': 1
    }
}

function scalrSignatureAlgorithm (keySecret, canonReq) {
    return format('V1-HMAC-SHA256 {}', crypto.createHmac('sha256', keySecret).update(canonReq).digest('base64'));
}

function scalrCanonicalRequest (method, date, path, params, body) {
    return [method, date, path, params, body].join('\n');
}

function scalrAPICall (creds, path, method, params, body, date, callback) {
    request.get({
        url: creds.api_url + path,
        headers: scalrRequestAuthentication(
                    creds.api_key_id, creds.api_key_secret,
                    method, path, params, body, date
                ),
        json: true
    }, function(error, response, body) {
        if (error) {
            callback(error, http);
        } else {
            callback(null, body);
        };
    });
}

var config = process.argv[1].replace(/\.js$/, '.json');
if (fs.existsSync(config)) {
    var date = new Date().toISOString();
    var creds = JSON.parse(fs.readFileSync(config).toString());
    scalrAPICall(creds, '/api/v1beta0/account/environments',
                'GET', '', '',  date, function(error, content) {
                    if (!error) {
                        console.log(content);
                    }
                });
}
